### PR TITLE
[keba] Add setenergy command

### LINKED ISSUE #4417: [keba] Add support for setting the charging energy limit
The Keba P30 Wallbox provides a feature to set the charging energy limit. This funktion is not accessible in the keba binding.

## Expected Behavior
With an Entry Field the changring limit in [W] can be entered. The value must be sent by a UDP command to the wallbox i.e. "setenergy 50000" for a limit of 5kWh. 

## Current Behavior
At the moment this feature is not available, but the new limit function works like the already existing "maxpresetcurrent" value. Just with the command described above.

### PR DESCRIPTION
**Pull request by tiptick**
introduce setenergy command (for openhab#4417) this is adding read and write item setenergy to set a charging limit in Wh. Quite handy command to limit the overall carge the car gets. if you only want to charge 20kWh set it to 20000. also added documentation.

Fixes #4417

### CODE REVIEW

File: bundles/org.openhab.binding.keba/src/main/resources/ESH-INF/thing/kecontact.xml [hunk 1]
```diff
@@ -245,4 +246,10 @@
 		<description>Session ID of the last charging session</description>
 		<state readOnly="true"></state>
 	</channel-type>
+	<channel-type id="setenergy">
```
Comment (wborn) [inline bundles/org.openhab.binding.keba/src/main/resources/ESH-INF/thing/kecontact.xml]: Maybe a better name for this channel would be something like `sessionlimit` or `sessioncharginglimit`? Those are properties of the Thing whereas `setenergy` seems to be an operation to modify the property of the Thing.

Comment (tiptick) [reply bundles/org.openhab.binding.keba/src/main/resources/ESH-INF/thing/kecontact.xml]: The name was choosen from the documentation of the manufacturer. But sure the name setenergy could be missleading as thies is really changing the behaviour ( the charging will be interrupted once the limit is reached). I will change it to `"setenergylimit"` , to indicate that this sets a limit and to stay close to the documentation. Charginglimit could be also missleading as the charging is accounted in kW but this is a energy limit for kWh. Does this sounds right?

File: bundles/org.openhab.binding.keba/README.md [hunk 2]
```diff
@@ -96,6 +98,7 @@ sitemap demo label="Main Menu"
 				Text item=KebaFailSafeCurrent label="Failsafe supply current [%.0f mA]"
 				Text item=KebaSessionEnergy label="Energy during current session [%.0f Wh]"
 				Text item=KebaTotalEnergy label="Energy during all sessions [%.0f Wh]"
+                Switch item=KebaSetEnergy label="Set charge energy" mappings=[0="off", 20000="20kWh"]
```
Comment (kaikreuzer) [inline bundles/org.openhab.binding.keba/README.md]: please replace spaces by tabs here to fix the alignment

File: bundles/org.openhab.binding.keba/src/main/resources/ESH-INF/thing/kecontact.xml [hunk 3]
```diff
@@ -245,4 +246,10 @@
 		<description>Session ID of the last charging session</description>
 		<state readOnly="true"></state>
 	</channel-type>
+	<channel-type id="setenergylimit">
+		<item-type>Number</item-type>
+		<label>Setenergylimit</label>
```
Comment (kaikreuzer) [inline bundles/org.openhab.binding.keba/src/main/resources/ESH-INF/thing/kecontact.xml]: Please use a proper label for the UI, e.g. "Energy Limit".

File: bundles/org.openhab.binding.keba/src/main/resources/ESH-INF/thing/kecontact.xml [hunk 4]
```diff
@@ -245,4 +246,10 @@
...
+		<description>set an energy limit for an already running or the next charging session.</description>
```
Comment (kaikreuzer) [inline bundles/org.openhab.binding.keba/src/main/resources/ESH-INF/thing/kecontact.xml]:
```suggestion
		<description>An energy limit for an already running or the next charging session.</description>
```

File: bundles/org.openhab.binding.keba/src/main/resources/ESH-INF/thing/kecontact.xml (hunk 1 above)
Comment (kaikreuzer) [reply bundles/org.openhab.binding.keba/src/main/resources/ESH-INF/thing/kecontact.xml]: Sounds right to me.

**Comment by kaikreuzer** [comment]
@tiptick Please do not do any merge commits into your PR - always rather do a rebase on master to update it.
Any chance that you do a squash of your commits as the history is quite dirty and the sign-offs are partially missing?

**Comment by tiptick** [comment]
changes from the PR review applied.
BUT: the merge commits i cannot get rid of, seems like git pulled some old commits onto the same branch. so in order to get again a clean history. please CLOSE this PR, i will tidy up my environment, and come back with a new PR,  stashing my changes.

**Comment by wborn** [comment]
Just rename your old branch, create a new one with the same name and after it's all nice and tidy force push it to your repo. Then the branch will be overwritten and this PR will also be nice and tidy without having to close it. ;-)

**Comment by tiptick** [comment]
@wborn thanks for the hint ! kindly check now, but DCO is happy, so am i :)

**Comment by wborn** [comment]
Great! Looks nice and tidy now. :smile:

**Pull request merged**